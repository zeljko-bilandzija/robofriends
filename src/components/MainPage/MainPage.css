@font-face {
    font-display: swap; /* This helps as improvement website performanse when loading fonts*/
    font-family: 'SEGA LOGO FONT';
    font-style: normal;
    font-weight: normal;
    src: local('SEGA LOGO FONT'), url('../../assets/fonts/SEGA.woff') format('woff');
}

h1 {
    font-family: 'SEGA LOGO FONT';
    font-weight: normal;
    color: #0ccac4;
}