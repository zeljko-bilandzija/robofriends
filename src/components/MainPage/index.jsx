import React, { Component } from 'react';
// import { connect } from 'react-redux';
import './MainPage.css';

import Header from '../Header';
import CardList from '../CardList';
import SearchBox from '../SearchBox';
import Scroll from '../../hoc/Scroll';
import ErrorBoundry from '../../hoc/ErrorBoundry';

export default class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }

    render() {
        const {              searchField, onSearchChange, isPending } = this.props;

        // In case of long loading
        if (isPending         ) {
            return <h1>        Loading...</h1>;
        }

        return (
            <div className="tc">
                <Header />
                <SearchBox searchField={searchField} onSearchChange={onSearchChange} />
                <Scroll>
                    <ErrorBoundry>
                        <CardList robots={this.filterRobots()} />
                    </ErrorBoundry>
                </Scroll>
            </div>
        );
    }

    filterRobots = () => {
        return this.props.robots.filter(robot => robot.name.toLowerCase().includes(this.props.searchField));
    }
};