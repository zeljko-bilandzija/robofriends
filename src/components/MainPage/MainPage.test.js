import React from 'react';
import MainPage from './index';
import { shallow } from 'enzyme';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    };
    wrapper = shallow(<MainPage {...mockProps} />);
});

it('should render MainPage component without crashing', () => {
    expect(wrapper).toMatchSnapshot();
});

it('should filters robots correctly', () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);

    const mockPropsWithRobot = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    };
    const wrapperWithRobot = shallow(<MainPage {...mockPropsWithRobot} />);
    expect(wrapperWithRobot.instance().filterRobots()).toEqual([{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
    }]);
});

it('should filters robots correctly 2', () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);

    const mockPropsWithRobot = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: 'a',
        isPending: false
    };
    const wrapperWithRobot = shallow(<MainPage {...mockPropsWithRobot} />);
    expect(wrapperWithRobot.instance().filterRobots()).toEqual([]);
});