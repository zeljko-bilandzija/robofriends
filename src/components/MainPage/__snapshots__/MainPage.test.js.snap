// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MainPage component without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    isPending={false}
    onRequestRobots={[Function]}
    robots={Array []}
    searchField=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <searchbox
          onSearchChange={undefined}
          searchField=""
/>,
        <scroll>
          <ErrorBoundry>
                    <cardlist
                              robots={Array []}
                    />
          </ErrorBoundry>
</scroll>,
      ],
      "className": "tc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onSearchChange": undefined,
          "searchField": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ErrorBoundry>
            <cardlist
                        robots={Array []}
            />
</ErrorBoundry>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <cardlist
              robots={Array []}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "robots": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <searchbox
            onSearchChange={undefined}
            searchField=""
/>,
          <scroll>
            <ErrorBoundry>
                        <cardlist
                                    robots={Array []}
                        />
            </ErrorBoundry>
</scroll>,
        ],
        "className": "tc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onSearchChange": undefined,
            "searchField": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ErrorBoundry>
              <cardlist
                            robots={Array []}
              />
</ErrorBoundry>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <cardlist
                robots={Array []}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "robots": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
